application/deed/views.py

```diff
@@ -106,3 +108,51 @@ def delete_borrower(borrower_id):
         abort(status.HTTP_404_NOT_FOUND)
     else:
         return jsonify({"id": borrower_id}), status.HTTP_200_OK
+
+
+@deed_bp.route('/<deed_reference>/sign', methods=['POST'])
+def sign_deed(deed_reference):
+    request_json = request.get_json()
+    borrower_token = request_json['borrower_token']
+    result = Deed.query.filter_by(token=str(deed_reference)).first()
+
+    if result is None:
+        LOGGER.error("Database Exception 404 for deed reference - %s" % deed_reference)
+        abort(status.HTTP_404_NOT_FOUND)
+    else:
+        LOGGER.info("Signing deed for borrower_id %s against deed reference %s" % (borrower_token, deed_reference))
+        # result.deed['token'] = result.token - what is the purpose of this? does this still need to be done when signing?
+        # check if XML already exisit
+        # if result.deed_xml is None:
+        #     deed_XML = convert_json_to_xml(result.deed)
+        #     print("deed_XML = %s" % deed_XML)
+        #     result.deed_xml = deed_XML.decode("UTF-8")
+        # sign then save to DB.
+
+        result.deed = update_deed_signature_timestamp(result, borrower_token)
+
+    return jsonify({"deed": result.deed}), status.HTTP_200_OK
+
+
+@deed_bp.route('/<deed_reference>/make-effective', methods=['POST'])
+def make_effective(deed_reference):
+    return status.HTTP_200_OK
+
+
+def update_deed_signature_timestamp(deed, borrower_token):
```

Comment:
push this function into the service.py file to keep the top level routes file as clean as possible pls